Sources/HNBUI/TabBar.swift

```diff
@@ -29,36 +35,34 @@ public final class TabBar: UIView {
         }
     }
 
-    public internal(set) var selectedItem: UITabBarItem? {
-        get {
-            guard let continuousIndex = continuousIndex else {
-                return nil
-            }
-            return items.count > 0 ? items[continuousIndex.roundedInt()] : nil
-        }
-        set {
-            guard let item = newValue, let index = items.firstIndex(of: item) else {
+    public private(set) var selectedItem: UITabBarItem? {
+        didSet {
+            guard let selectedItem = selectedItem,
+                  let index = items.firstIndex(of: selectedItem),
+                  let tabBarItemViews = hStack.arrangedSubviews as? [TabBarItemView]
+             else {
                 continuousIndex = nil
                 return
             }
-            setContinuousIndex(CGFloat(index), animated: true)
+            tabBarItemViews.forEach { tabBarItemView in
+                tabBarItemView.isSelected = false
+            }
+            tabBarItemViews[index].isSelected = true
+            delegate?.tabBar(self, didSelectItem: selectedItem, atIndex: index)
         }
     }
 
     private var continuousIndex: CGFloat? {
         didSet(oldValue) {
-            guard continuousIndex?.roundedInt() != oldValue?.roundedInt(),
-                  let tabBarItems = hStack.arrangedSubviews as? [TabBarItemView],
-                  let continuousIndex = continuousIndex,
+            guard let continuousIndex = continuousIndex,
                   continuousIndex >= 0,
-                  Int(continuousIndex) < tabBarItems.count else {
+                  Int(continuousIndex) < hStack.arrangedSubviews.count,
+                  continuousIndex.truncatingRemainder(dividingBy: 1) == 0.0,
+                  selectedItem != items[Int(continuousIndex)]
+            else {
                 return
             }
-            tabBarItems.forEach { tabBarItem in
-                tabBarItem.isSelected = false
-            }
-            tabBarItems[continuousIndex.roundedInt()].isSelected = true
-            delegate?.tabBar(self, didSelectItem: items[continuousIndex.roundedInt()], atIndex: continuousIndex.roundedInt())
+            selectedItem = items[Int(continuousIndex)]
```

Comment:
selectedItem is updated when 
`continuousIndex` is within the range of `hStack.arrangedSubviews.count`
`continuousIndex` does not have decimal
already selectedItem and new item are different.

This means item is selected when view's transition is completed.
Before, item was selected when the transition is go over the half point.